robot/client/client.go

```diff
@@ -117,17 +117,18 @@ type RobotClient struct {
 	address     string
 	dialOptions []rpc.DialOption
 
-	mu              sync.RWMutex
-	resourceNames   []resource.Name
-	resourceRPCAPIs []resource.RPCAPI
-	resourceClients map[resource.Name]resource.Resource
-	remoteNameMap   map[resource.Name]resource.Name
-	changeChan      chan bool
-	notifyParent    func()
-	conn            reconfigurableClientConn
-	client          pb.RobotServiceClient
-	refClient       *grpcreflect.Client
-	connected       atomic.Bool
+	mu                    sync.RWMutex
+	resourceNames         []resource.Name
+	resourceRPCAPIs       []resource.RPCAPI
+	resourceClients       map[resource.Name]resource.Resource
+	remoteNameMap         map[resource.Name]resource.Name
+	changeChan            chan bool
+	notifyParent          func()
+	conn                  reconfigurableClientConn
+	client                pb.RobotServiceClient
+	refClient             *grpcreflect.Client
+	connected             atomic.Bool
+	subtypesUnimplemented bool
```

Comment:
[opt] might want to use an atomic.bool here in case this is ever referenced concurrently. 

Lots of things in this file are called simultaneously.

Reply:
Agree with @mattjperez and generally think making `bool`s `atomic.Bool`s with the worry that they will be accessed concurrently can be misleading. I think a `bool` should only be `atomic.Bool` if it's definitely accessed concurrently; otherwise future developers will assume it _is_ accessed concurrently and the code becomes less readable.